libs/libc/time/lib_localtime.c

```diff
@@ -307,14 +307,15 @@ struct rule_s
 
 /* The minimum and maximum finite time values.  */
 
-static const time_t g_min_timet =
-  (TYPE_SIGNED(time_t)
-    ? (time_t)-1 << (CHAR_BIT * sizeof(time_t) - 1)
+static const int_fast64_t g_min_timet =
+  (TYPE_SIGNED(int_fast64_t)
+    ? (int_fast64_t)1 << (CHAR_BIT * sizeof(int_fast64_t) - 1)
```

Comment:
why switched from `-1` to `1` here?

Reply:
CI report Error:
Error: time/lib_localtime.c:312:24: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    ? (int_fast64_t)-1 << (CHAR_BIT * sizeof(int_fast64_t) - 1)
      ~~~~~~~~~~~~~~~~ ^
Error: time/lib_localtime.c:317:39: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    ? - (~ 0 < 0) - ((int_fast64_t)-1 <<
                     ~~~~~~~~~~~~~~~~ ^